helix-term/src/commands/typed.rs

```diff
@@ -2408,6 +2408,57 @@ fn redraw(
     Ok(())
 }
 
+fn move_buffer(
+    cx: &mut compositor::Context,
+    args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+
+    ensure!(args.len() == 1, format!(":move takes one argument"));
+
+    let new_path = args.first().unwrap().to_string();
+    let doc = doc!(cx.editor);
+    let old_path = doc
+        .path()
+        .ok_or_else(|| anyhow!("Scratch buffer cannot be moved. Use :write instead"))?
+        .to_string_lossy()
+        .to_string();
+
+    let lsp_reply = doc.language_servers().find_map(|lsp| {
+        if let Some(future) = lsp.prepare_file_rename(old_path.clone(), new_path.clone()) {
+            match helix_lsp::block_on(future) {
+                Ok(edit) => return Some(edit),
+                Err(e) => {
+                    log::error!("LSP willRename request failed: {:?}", e);
+                }
+            }
+        };
+        None
+    });
```

Comment:
I am not really sure if this is the correct approach for handling `will_rename`. You likely want to call `will_rename` for all servers. Workspace edits will not be applied if they conflict with previous changes anyway.

I think some servers would not handle it well if we called `did_rename` without `will_rename`.

Reply:
Yep, I would tend to agree as well, but I followed that past comment of yours :smile: 
https://github.com/helix-editor/helix/pull/5531/files#r1255798844

Code is updated, let me know if there's way to make it more readable (I'm not that good with rust idoms) :pray:

Reply:
yeah back then the idea was that we would only send both `willRename` and `didRename` to a single server but I support sending it to all of them should really work aswell.